components/tag-list/tag-list.js

```diff
@@ -222,17 +266,57 @@ class TagList extends LocalizeCoreElement(ArrowKeysMixin(LitElement)) {
 		return items.filter(({ isHidden }) => !isHidden);
 	}
 
-	_getTagListItems() {
+	async _getTagListItems() {
 		const slot = this.shadowRoot && this.shadowRoot.querySelector('slot');
 		if (!slot) return;
-		return slot.assignedNodes({ flatten: true }).filter((node) => {
+
+		const results = await filterAsync(slot.assignedNodes({ flatten: true }), async node => {
 			if (node.nodeType !== Node.ELEMENT_NODE) return false;
+			await node.updateComplete;
+
 			const role = node.getAttribute('role');
+			if (role !== 'listitem') return false;
+
+			if (this.clearable) node.setAttribute('clearable', 'clearable');
+			node.addEventListener('d2l-tag-list-item-cleared', this._handleItemDeleted.bind(this));
```

Comment:
Do we need to wire up the listener to the tag element?  Can we wire up the handler once above in `render` to the tag list container?  It would ensure that we never wire up more than one handler for removing the tag.  It also ensures that the handler runs in the correct scope.  

I know this is a contrived example because we have no intentions of supporting drag & drop of tags, but if a tag were drag & dropped from one tag list to another and then cleared, it looks like a second handler would get wired up but the original one would still run against the original scope. I think it just highlights some of the pitfalls wiring up things across scopes.

Reply:
Yup that works as well and does sound like a safer option.